packages/vite-plugin-cloudflare/src/assert-wrangler-version.ts

```diff
@@ -0,0 +1,36 @@
+import { satisfies } from "semver";
+
+/**
+ * Asserts that the installed version of Wrangler that gets pulled in at runtime by the `@cloudflare/vite-plugin`
+ * matches the version that `@cloudflare/vite-plugin` actually depends upon.
+ *
+ * This can sometime be broken by package managers that deduplicate dependencies, such as `pnpm`.
+ */
+export async function assertWranglerVersion() {
+	const installedVersion = (
+		await import("wrangler/package.json", {
+			with: { type: "json" },
+		})
+	).default.version;
+
+	const ourPackageJson = (
+		await import("../package.json", {
+			with: { type: "json" },
+		})
+	).default;
```

Comment:
Why does this need to be a dynamic import? What about just importing directly (and the value being bundled in)